backend/src/auth-service/models/mail.go

```diff
@@ -0,0 +1,134 @@
+package models
+
+import (
+	"crypto/tls"
+	"fmt"
+	"log"
+	"net/smtp"
+
+	"github.com/astaxie/beego"
+	"github.com/astaxie/beego/config"
+)
+
+type MailConfig struct {
+	SmtpHttpsPort  string
+	SmtpUrl        string
+	SenderEmail    string
+	SenderPassword string
+}
+
+var mailConfig MailConfig
+
+type Mail struct {
+	senderId string
+	toId     string
+	subject  string
+	body     string
+	pass     string
+}
+
+type SmtpServer struct {
+	host string
+	port string
+}
+
+func init() {
+	mailconf, err := config.NewConfig("ini", "conf/mail.conf")
+	if err != nil {
+		beego.Critical(err.Error())
+		panic(err)
+	}
+	mailConfig = MailConfig{
+		SmtpHttpsPort:  mailconf.String("https_port"),
+		SmtpUrl:        mailconf.String("smtp_host"),
+		SenderEmail:    mailconf.String("sender_email"),
+		SenderPassword: mailconf.String("sender_password"),
+	}
+
+}
+
+func (s *SmtpServer) ServerName() string {
+	return s.host + ":" + s.port
+}
+
+func (mail *Mail) BuildMessage() string {
+	message := ""
+	message += fmt.Sprintf("From: %s\r\n", mail.senderId)
+	if len(mail.toId) > 0 {
+		message += fmt.Sprintf("To: %s\r\n", mail.toId)
+	}
+
+	message += fmt.Sprintf("Subject: %s\r\n", mail.subject)
+	message += "\r\n" + mail.body + mail.pass
+
+	return message
+}
+
+func SendingRegistrationToken(pass string, Login string) *error {
+
+	//build a message
+	mail := Mail{}
+
+	mail.senderId = mailConfig.SenderEmail
+	mail.toId = Login
+	mail.subject = "StudIT registration"
+	mail.body = "Copy this registration code:\n\n"
+	mail.pass = pass
+
+	messageBody := mail.BuildMessage()
+	smtpServer := SmtpServer{host: mailConfig.SmtpUrl, port: mailConfig.SmtpHttpsPort}
+
+	log.Println(mailConfig)
+	//build an auth
+	auth := smtp.PlainAuth("", mail.senderId, mailConfig.SenderPassword, smtpServer.host)
+
+	// Gmail will reject connection if it's not secure
+	// TLS config
+	tlsconfig := &tls.Config{
+		InsecureSkipVerify: true,
+		ServerName:         smtpServer.host,
+	}
+
+	conn, err := tls.Dial("tcp", smtpServer.ServerName(), tlsconfig)
+	if err != nil {
+		beego.Debug("Register error:" + err.Error())
```

Comment:
Why "Register error"?

Reply:
In case of error, the method should return this error. But now, there is the only `beego.Debug(...)`